Subscription Storage

catzkorn opened the pull request.

Commit:
Add database setup file

- Create db/migrations folder structure

- Create database_setup.sql with schema in

Commit:
Add NewPostgresStore method

- NewPostgresStore starts the connection with the specified database

- NewPostgresStore takes a DSN parameter, to allow the user to specifiy the database

Commit:
Update NewPostgresStore to NewSubscriptionStore

- NewSubscriptionStore starts the connection with the specified database, and checks if there is a connection via .Ping

- Add RecordSubscriptions method and test, which inserts a subscription to the database

- Add GetSubscriptions method and tests, which retrieves subscriptions from the database

veronicavioletlee commented on SubscriptionStore_test.go:
```diff
@@ -0,0 +1,94 @@
+	t.Run("tests a successful database connection", func(t *testing.T) {
+		_, err := NewSubscriptionStore("user=charlotte  host=localhost port=5432 database=subscryptdb sslmode=disable")
+
+		if err != nil {
```
We could look at separating the error checks to separate methods for re-use in the tests

0xfarhaan commented on SubscriptionStore.go:
```diff
@@ -0,0 +1,82 @@
+}
+
+// NewSubscriptionStore starts connection with database
+func NewSubscriptionStore(databaseDSN string) (*SubscriptionStore, error) {
```
rename function to something like "NewDatabaseConnection()"

Commit:
Refactor SubscriptionStore and SubscriptionStore_Test

- NewSubscriptionStore renamed to NewDatabaseConnection

- Add assertDatabaseError test helper method

- Rename test from TestDatabaseExecContext to TestDatabaseFunctionality

Commit:
Merge main into postgres

- Fix date handling in server_test and server_integration_test

Commit:
Add database intergration test

- Fix methods to return errors

- Test passes for InMemoryStore and Database

Commit:
Add environmental variables

- Add .env file to contain database connection string

- Add GetDatabaseEnvVariable method that loads the .env file and gets the environmental variable

Commit:
Update .travis.yml

- Add services: postgresql

- Add env: DATABASE_CONN_STRING

- Add before_script
